### PR TITLE
add mutation webhook controller

### PR DESCRIPTION
**Pull request by achrefbensaad**
Remove annotations logic from kubearmor
update ci.
Signed-off-by: Achref Ben Saadd <achref9612@gmail.com>

### CODE REVIEW

**Comment by achrefbensaad** [comment]
need to be rebased before merge

**Comment by achrefbensaad** [comment]
@nyrahul @nam-jaehyun @daemon1024 Please review this

File: .github/workflows/ci-test.yml [hunk 1]
```diff
@@ -56,6 +56,34 @@ jobs:
         run: |
           ./contribution/k3s/install_k3s.sh
 
+      - name: Setup self-hosted docker registry
+        run: |
```
Comment (daemon1024) [inline .github/workflows/ci-test.yml]: Can we move these to scripts instead? We would be needing them local dev environment too right? Like how we are calling k3s installation script.

Comment (achrefbensaad) [reply .github/workflows/ci-test.yml]: Yes, its a good idea. I have implimented the required changes. Please take an other look.

**Review by daemon1024 — APPROVED** [review]

**Comment by nam-jaehyun** [comment]
1. I'd like to still keep the code in kubeUpdate.go since the admission controller only works in the cluster where all nodes are based on ubuntu. Also, we need to test KubeArmor in a local env (e.g., make run), but the admission controller might be not in such an env.
2. I don't get why we need to create a local registry, which increases the overall test complexity.

**Review by nam-jaehyun — CHANGES_REQUESTED** [review]
Multiple files are specifically updated for the admission controller, but the updated files are also used for some other purposes. We may need to think of this PR again.

**Comment by nam-jaehyun** [comment]
refer to https://github.com/kubearmor/KubeArmor/pull/690

**Comment by nam-jaehyun** [comment]
outdated PR. we have a separate PR for the CI update (#695).

**Pull request closed without merging**